File: Profiletest/reversestring.java
public class reversestring{
    public static String reverse(String str){
        if(str.length() == 0)
            return str;
        return reverse(str.substring(1)) + str.charAt(0);
    }
    public static void main(String[] args) {
        String str1 = "Shivam Thakur";
        System.out.println(reverse(str1));
    }
}

File: day3/reversestring.c
#include<stdio.h>
void reverse(char arr[], int i){
    if(arr[i] == '\0') return;
    reverse(arr, i+1);
    printf("%c", arr[i]);
}
int main(){
    char str[] = "shivam Thakur";
    reverse(str, 0);
    return 0;
}